imports/src/parser/parse_symbol.rs

```diff
@@ -23,99 +23,118 @@ use std::{ffi::OsString, fs::DirEntry, path::PathBuf};
-    let file_name = entry
+        .map_err(|error| ImportParserError::directory_error(error, span.clone(), &package.path()))?;
+    let file_name = package
         .file_name()
         .into_string()
-        .map_err(|_| ImportError::convert_os_string(span.clone()))?;
+        .map_err(|_| ImportParserError::convert_os_string(span.clone()))?;
 
-    let mut file_path = entry.path();
+    let mut file_path = package.path();
     if file_type.is_dir() {
         file_path.push(LIBRARY_FILE);
 
         if !file_path.exists() {
-            return Err(ImportError::expected_lib_file(
+            return Err(ImportParserError::expected_lib_file(
                 format!("{:?}", file_path.as_path()),
                 span.clone(),
             ));
         }
     }
 
-    // Builds the abstract syntax tree.
+    // Build the package abstract syntax tree.
     let program_string = &LeoAst::load_file(&file_path)?;
     let ast = &LeoAst::new(&file_path, &program_string)?;
 
-    // Generates the Leo program from file.
+    // Build the package typed syntax tree from the package abstract syntax tree.
     Ok(Program::from(&file_name, ast.as_repr()))
 }
 
 impl ImportParser {
-    pub fn parse_import_star(&mut self, entry: &DirEntry, span: &Span) -> Result<(), ImportError> {
-        let path = entry.path();
+    ///
+    /// Import all symbols from a given package.
+    ///
+    /// If the package is a Leo file, import all symbols from the file.
+    /// If the package is a directory, import all symbol from the library file.
+    ///
+    pub fn parse_import_star(&mut self, package: &DirEntry, span: &Span) -> Result<(), ImportParserError> {
+        let path = package.path();
         let is_dir = path.is_dir();
+
+        // Check if the package is a Leo file.
         let is_leo_file = path
             .extension()
             .map_or(false, |ext| ext.eq(&OsString::from(FILE_EXTENSION)));
 
         let mut package_path = path;
         package_path.push(LIBRARY_FILE);
 
+        // Check if the package is a directory.
         let is_package = is_dir && package_path.exists();
 
         // import * can only be invoked on a package with a library file or a leo file
         if is_package || is_leo_file {
-            // Generate aleo program from file
-            let program = parse_import_file(entry, &span)?;
+            // Get the package typed syntax tree.
+            let program = parse_import_file(package, span)?;
 
-            // Store program's imports in imports hashmap
+            // Insert the package's imports into the import parser.
             program
                 .imports
                 .iter()
-                .map(|import| self.parse_package(entry.path(), &import.package))
-                .collect::<Result<Vec<()>, ImportError>>()?;
+                .map(|import| self.parse_package(package.path(), &import.package))
+                .collect::<Result<Vec<()>, ImportParserError>>()?;
```

Comment:
as before, this should instead be a plain `for` loop, so that no allocation occurs in the happy path